modules/certificate/certificate.service.js

```diff
@@ -67,17 +79,23 @@ class CertificatesService {
   }
 
   async generateCertificate(certificateData, userId, userRole) {
-    const randomNum = randomInt(100000, 999999);
-
-    certificateData.name = `hor${randomNum}`;
+    const certificatesArr = [];
+    const { count } = certificateData;
 
     if (userRole === USER) {
       certificateData.ownedBy = userId;
     } else {
       certificateData.createdBy = userId;
     }
 
-    return new CertificateModel(certificateData).save();
+    for (let i = 0; i < count; i++) {
+      certificateData.name = await generateName();
+      certificatesArr.push({ ...certificateData });
+    }
+
+    const certificates = await CertificateModel.insertMany(certificatesArr);
+
+    return { certificates };
   }
```

Comment:
I think generateCertificate method needs to create only one Certificate. 
How about to get a query with an object {userId, userRole, [certificateData1, certificateData2, ...]} that will be processed by another method that will invoke generateCertificate and then returns all generated certificates?
```suggestion
  }
  async someMethod (userId,userRole, certArray) {
const  generatedCerts = certArray.map(item => await this.generateCertificate(userId,userRole, item) )
return generatedCerts;
```

Reply:
I think we can leave it as it is, because at the front and on the admin we always specify the number of certificates